sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java

```diff
@@ -90,7 +148,35 @@ public void initPassword(
       String userId,
       PasswordChangeRequest pcr,
       List<SendMode> sendMode) {
-    boolean newPasswordIsValid = passwordService.validatePassword(pcr.getNewPassword());
+    Map<String, String> usProperties =
+        userStorage != null
+            ? configService.getRealm(realm).getUserStorages().stream()
+                .filter(u -> u.getName().equalsIgnoreCase(userStorage))
+                .findFirst()
+                .get()
+                .getProperties()
+            : configService.getRealm(realm).getUserStorages().stream()
+                .findFirst()
+                .get()
+                .getProperties();
+    boolean newPasswordIsValid =
+        passwordService.validatePassword(
+            pcr.getNewPassword(),
+            usProperties.get("create_password_withUpperCase") != null
```

Comment:
I think those key strings should be declared as constants to avoid error in the future